torch/csrc/jit/passes/canonicalize_ops.cpp

```diff
@@ -62,6 +62,21 @@ static void CanonicalizeOps(Block* block) {
 
       it->output()->replaceAllUsesWith(result);
       it.destroyCurrent();
+    } else if (it->matches("aten::add(Tensor self, Tensor other, *, Scalar alpha) -> Tensor") ||
+               it->matches("aten::sub(Tensor self, Tensor other, *, Scalar alpha) -> Tensor") ||
+               it->matches("aten::mul(Tensor self, Tensor other) -> Tensor") ||
+               it->matches("aten::div(Tensor self, Tensor other) -> Tensor")) {
+      if (auto other = it->get<at::Tensor>(attr::other)) {
+        if (other->dim() == 0) {
+          WithInsertPoint insert_guard {*it};
+          auto graph = it->owningGraph();
+          auto new_other = graph->insertConstant(other->_local_scalar());
```

Comment:
I think this code is taking a scalar tensor and inlining its value into the graph -- what if the scalar tensor isn't constant?

Reply:
`it->get<...>(...)` will return a value only if it's a constant, so this is valid.